### PR TITLE
Job management

### PR DESCRIPTION
**Pull request by minotogna**
### Job Management 

- It's a typescript conversion of what implemented in Arena
- Added tests
- TODO: WebSocket management (separate task)

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-gk4tum Job management ](https://app.clickup.com/t/gk4tum)

File: src/db/dbMigrator/dbMigrator.ts [hunk 1]
```diff
@@ -10,13 +11,15 @@ const logger = new Logger('DBMigrator')
 const migrateSchema = async (params: { schema?: string; migrationsFolder?: string } = {}): Promise<void> => {
   const { schema = Schemata.PUBLIC, migrationsFolder = __dirname } = params
 
-  if (schema !== Schemata.PUBLIC) {
-    await DB.none(`CREATE SCHEMA IF NOT EXISTS ${schema}`)
-  }
+  if (fs.existsSync(migrationsFolder)) {
```
Comment (minotogna) [inline src/db/dbMigrator/dbMigrator.ts:14]: this is to avoid exception in case migrationsMolder do not exist

File: src/job/job.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  new (data: any, jobs?: Array<JobServer<any, any, any>>): JobServer<any, any, any>
+  new <P extends JobData, R, C extends JobContext>(data: P, jobs?: Array<JobServer<P, R, C>>): JobServer<P, R, C>
+  readonly prototype: JobServer<any, any, any>
+}
+
+export abstract class JobServer<P extends JobData, R, C extends JobContext> extends EventEmitter implements Job<R> {
+  summary: JobSummary<R>
+  protected readonly logger = new Logger(`Job ${this.constructor.name}`)
+  protected context: C
+  protected workerData: P
```
Comment (minotogna) [inline src/job/job.ts:22]: @SteRiccio should we merge workerData (ex. params) and context into context ?

File: src/db/dbMigrator/dbMigrator.ts (hunk 1 above)
Comment (sorja) [inline src/db/dbMigrator/dbMigrator.ts:14]: Should we avoid nested ifs?
`if (!fs.existsSync(migrationsFolder)) return`

File: src/server/error.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  SERVICE_UNAVAILABLE = 503,
+  GATEWAY_TIMEOUT = 504,
+}
+
+export class ServerError extends Error {
+  key: string
+  params: any
+  statusCode: ServerErrorCode
+
+  constructor(key: string, params?: any, statusCode: ServerErrorCode = ServerErrorCode.INTERNAL_SERVER_ERROR) {
```
Comment (sorja) [inline src/server/error.ts:27]: Should `statusCode` be optional?

Comment (minotogna) [reply src/server/error.ts:27]: it is optional. it takes  ServerErrorCode.INTERNAL_SERVER_ERROR if not passed.
or I misunderstood the comment?

File: src/db/dbMigrator/dbMigrator.ts (hunk 1 above)
Comment (minotogna) [reply src/db/dbMigrator/dbMigrator.ts]: yes, see my last commit

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**